Calculator homework

TheCaerus opened the pull request:
Unfinished homework - will perform fully till this Friday.

TheCaerus committed:
Create calculator-homework.spec.js

TheCaerus committed:
Updated

DominykasPoskus1 reviewed (changes requested):
Hey, short feedback that could help you with test updates:
- Create some helper functions on calculatorStartPage.js instead of filling every field by using selectOption, click and other functions separately.
- Test all build versions with the same test list and data for each separate tests run so that you know that data setup was the same when some issues occur.
- Don't forget that there are some more functionality than just math functions (clear button and so on) - test them too.
- Good job using random numbers instead of hard coded ones

TheCaerus committed:
Updated version of homework

More automated testing scenarios provided. Didn't managed to distribute to classes.

DominykasPoskus1 reviewed (changes requested):
Hello,
Overall, really good homework, good job:) I see you went an extra mile with descriptions and titles, looks nice.

Comments:
- I like foreaches - but maybe you could refactor them to one instead of many.
- You have touched page object file with initiateFill(), why not to move more methods there what are repetitive?
- I see that no tests check the integer checkbox or validation errors.
- You could try math functions with float numbers as well.
